teuthology/suite/run.py

```diff
@@ -359,8 +359,10 @@ def collect_jobs(self, arch, configs, newest=False):
                     'Stopped after {limit} jobs due to --limit={limit}'.format(
                         limit=limit))
                 break
-            # Break apart the filter parameter (one string) into comma
-            # separated components to be used in searches.
+            filter_all = self.args.filter_all
+            if filter_all:
+                if not all(_ in description for _ in filter_all):
```

Comment:
I'm bad at list comprehensions so I'm having trouble telling if this will work...but shouldn't it be bracketed like the others?

Reply:
In this case it is not required, without brackets python creates iterator, the all can take utterable objects, so, it is more reasonable to use it not wasting the memory. I know we are not struggle for the performance in our scripting, but this form is legitimate.

Reply:
Example:
```
>>> a=[1, 2, 3]
>>> b=[2, 2, 2]
>>> all(_==2 for _ in a)
False
>>> all(_==2 for _ in b)
True

```

Reply:
👍

Reply:
i'd suggest `s/_/x/`, because 

1. we are not ignoring `x`
2. for the sake of consistency.

Reply:
@tchaikov nevertheless someone uses underscore as ignore variables, another good usage example is to utilize it for looping and iterators, since the underscore is the same variable as `x`, or `i` or other, does not have a specific naming, for better abstraction. Secondly, it improves readability especially in comprehensions, we have a lot of example how to do not right code like the one brightest `[param for param in params where param]` :-)
Though, I agree we can try to keep naming consistency.

Reply:
> for better abstraction

i don't think we need abstraction here. as it's a concrete data structure -- a list of keywords. instead of a general purpose data structure.

>  it improves readability especially in comprehensions

i am not convinced regarding to the readability improvements. if `_` had a name, it helps me to understand what it presents.

Reply:
> > for better abstraction
> 
> i don't think we need abstraction here. as it's a concrete data structure -- a list of keywords. instead of a general purpose data structure.
> 
> > it improves readability especially in comprehensions
> 
> i am not convinced regarding to the readability improvements. if `_` had a name, it helps me to understand what it presents.

I meant here that abstract context variable name is better for me, I don't care about which exactly name is better x, i, or anything less, this is why '_' is my choice, it is just more readabl, python's `_`  is like wildcard operator in other scripting languages with functional programming abilities,  `$_` in perl, `it` in groovy, `_` in scala.